paddle/fluid/operators/reduce_ops/reduce_op.h

```diff
@@ -559,15 +559,43 @@ class ReduceGradOp : public framework::OperatorWithKernel {
  protected:
   framework::OpKernelType GetExpectedKernelType(
       const framework::ExecutionContext& ctx) const override {
+    auto input_data_type = OperatorWithKernel::IndicateVarDataType(
+        ctx, framework::GradVarName("Out"));
+
+#ifdef PADDLE_WITH_MKLDNN
+    auto CanMKLDNNReduceGradBeUsed = [&]() {
+      auto dx_dims = ctx.Input<Tensor>("X")->dims();
+      if (ctx.Attr<bool>("reduce_all") ||
+          ((int)ctx.Attr<std::vector<int>>("dim").size() == dx_dims.size()))
+        return true;
+
+      if (dx_dims.size() > 5) return false;  // max 5D tensor is supported
```

Comment:
With this second condition `Attr<std::vector<int>>("dim").size() == dx_dims.size()` you may pass reducing tensor with rank > 5 if one will pass manually reduction of all dims in "dim" attribute. Please first check for 5dim.